test/unit/validators/uri_validator_test.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class UriValidatorTest < ActiveSupport::TestCase
+  test 'invalid uri' do
+    klass = Class.new(base_record_class) { validates :uri, uri: true }
```

Comment:
Do you need to subclass the base_record?
Can't you use a fix `Model` class? Note it will be namespaced in `UriValidatorTest::Model`